rust/blockstore/src/arrow/block/delta/ordered_block_delta.rs

```diff
@@ -113,38 +100,43 @@ impl OrderedBlockDelta {
     }
 
     fn copy_up_to<'a, K: ArrowReadableKey<'a>, V: ArrowReadableValue<'a>>(
-        &mut self,
+        &'a mut self,
         excluded_prefix: &str,
-        excluded_key: KeyWrapper,
-        old_block: &'a Block,
+        excluded_key: &KeyWrapper,
     ) {
-        let prefix_arr = old_block
-            .data
-            .column(0)
-            .as_any()
-            .downcast_ref::<StringArray>()
-            .unwrap();
-        let key_arr = old_block.data.column(1);
-
-        for i in self.copied_up_to_row_of_old_block..old_block.data.num_rows() {
-            let old_prefix = prefix_arr.value(i);
-            let old_key = K::get(key_arr, i);
-            let old_key_wrapped: KeyWrapper = old_key.clone().into(); // todo: remove clone
-
-            if old_prefix > excluded_prefix
-                || (old_prefix == excluded_prefix && old_key_wrapped >= excluded_key)
-            {
-                // If the provided prefix/key pair existed in the old block, we need to skip it (move the pointer past that entry)
-                if old_prefix == excluded_prefix && old_key_wrapped == excluded_key {
-                    self.copied_up_to_row_of_old_block += 1;
+        if let Some(old_block) = self.old_block.as_ref() {
```

Comment:
For my understanding, why did we clone it previously?

Reply:
I had a hard time getting lifetimes to work out without cloning it before, turns out the solution was pretty simple.